src/redux/books/booksSlice.js

```diff
@@ -1,19 +1,39 @@
+/* eslint-disable no-param-reassign */
```

Comment:
Hi @emmiiorji, thanks for the feedback. I disabled the no-param-reassign linter check due to how Immer uses this syntax to reassign the state variable. I don't see any other alternative than this Immer syntax using redux toolkit. I would appreciate if you can be more specific about the recommendation as a viable alternative. Tks!

Reply:
Hi @dennis-every

You're quite correct about how createSlice would normally violate the no-para-reassign rule since it modifies the state object passed in as a parameter.

To use createSlice without disabling no-param-reassign, you can use destructuring to create a new object based on the state parameter and return it from the reducer function.

e.g
 ```javascript
const booksSlice = createSlice({
  name: 'books',
  initialState,
  reducers: {
    addBook: (state, action) => ({ ...state, books: [...state.books, action.payload] }),
    removeBook: (state, action) => ({
      ...state, books: state.books.filter((book) => book.item_id !== action.payload),
    }),
  },
});
addBook: (state, action) => ({ ...state, books: [...state.books, action.payload] })
```

Reply:
Hi @emmiiorji , here's a link from the official redux-toolkit docs explaining the eslint override: https://redux-toolkit.js.org/usage/immer-reducers#linting-state-mutations. Should I not follow the official docs in this case?

Reply:
I have already seen reasons with you and totally agree. But the danger in your code is that you have disabled it for the whole file which might let you miss out in detecting other occurrences.

If you must disable, I think you should use disable-next-line instead.

Reply:
Hi @emmiiorji , ok, tks for the explanation!